src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -3847,70 +3800,199 @@ int SketchObject::trim(int GeoId, const Base::Vector3d& point)
+                                                              cutPoints[1])
+                                        .release());
     }
+}
+
+int SketchObject::trim(int GeoId, const Base::Vector3d& point)
+{
+    if (!isGeoIdAllowedForTrim(this, GeoId)) {
+        return -1;
+    }
+    // Remove internal geometry beforehand for now
+    // FIXME: we should be able to transfer these to new curves smoothly
+    // auto geo = getGeometry(GeoId);
+    const auto* geoAsCurve = static_cast<const Part::GeomCurve*>(getGeometry(GeoId));
+
+    //******************* Step A => Detection of intersection - Common to all Geometries
+    //****************************************//
+    // GeoIds intersecting the curve around `point`
+    std::array<int, 2> cuttingGeoIds {GeoEnum::GeoUndef, GeoEnum::GeoUndef};
+    // Points at the intersection
+    std::array<Base::Vector3d, 2> cutPoints;
+
+    // Using SketchObject wrapper, as Part2DObject version returns GeoId = -1 when intersection not
+    // found, which is wrong for a GeoId (axis). seekTrimPoints returns:
+    // - For a parameter associated with "point" between an intersection and the end point
+    // (non-periodic case) cuttingGeoIds[0] != GeoUndef and cuttingGeoIds[1] == GeoUndef
+    // - For a parameter associated with "point" between the start point and an intersection
+    // (non-periodic case) cuttingGeoIds[1] != GeoUndef and cuttingGeoIds[0] == GeoUndef
+    // - For a parameter associated with "point" between two intersection points, cuttingGeoIds[0]
+    // != GeoUndef and cuttingGeoIds[1] != GeoUndef
+    //
+    // FirstParam < point1param < point2param < LastParam
+    if (!SketchObject::seekTrimPoints(GeoId,
+                                      point,
+                                      cuttingGeoIds[0],
+                                      cutPoints[0],
+                                      cuttingGeoIds[1],
+                                      cutPoints[1])) {
+        // If no suitable trim points are found, then trim defaults to deleting the geometry
+        delGeometry(GeoId);
+        return 0;
+    }
+
+    // TODO: find trim parameters
+    std::vector<std::pair<double, double>> paramsOfNewGeos;
+    paramsOfNewGeos.reserve(2);
+    if (!getParamLimitsOfNewGeosForTrim(this, GeoId, cuttingGeoIds, cutPoints, paramsOfNewGeos)) {
+        return -1;
+    }
+
+    //******************* Step B => Creation of new geometries
+    //****************************************//
+    std::vector<int> newIds;
+    std::vector<Part::Geometry*> newGeos;
+
+    switch (paramsOfNewGeos.size()) {
+        case 0: {
+            delGeometry(GeoId);
+            return 0;
+        }
+        case 1: {
+            newIds.push_back(GeoId);
+            break;
+        }
+        case 2: {
+            newIds.push_back(GeoId);
+            newIds.push_back(getHighestCurveIndex() + 1);
+            break;
+        }
+        default: {
+            return -1;
+        }
+    }

```

Comment:
No need for `{}` in the cases as there are no variables defined here.
```suggestion
    switch (paramsOfNewGeos.size()) {
        case 0:
            delGeometry(GeoId);
            return 0;
        case 1:
            newIds.push_back(GeoId);
            break;
        case 2:
            newIds.push_back(GeoId);
            newIds.push_back(getHighestCurveIndex() + 1);
            break;
        default:
            return -1;
    }
```

Reply:
I would still like to keep these unless explicitly forbidden somewhere. Mainly a matter of taste, but also spacemacs doesn't fold it otherwise, and the braces make these chunks easy to copy/move with Vi(m) bindings.

Reply:
From what I can see, the code base seems to lean towards skipping {} when there's no need for them (and using them for all cases when there's a need in one or more places).

I'd prefer if same style is used everywhere, but if the linter doesn't complain, then I won't either.
Either way it isn't something that should hold up the PR. The comment was mostly to inform you if you'd missed it :)